src/metax_api/tests/api/rest/base/views/directories/read.py

```diff
@@ -302,26 +332,53 @@ def test_read_directory_for_catalog_record_directory_does_not_exist(self):
         # should be OK...
         response = self.client.get('/rest/directories/4/files')
         self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.data['directories']), 1)
+        self.assertEqual(len(response.data['files']), 5)
 
         # ... but should not contain any files FOR THIS CR
         response = self.client.get('/rest/directories/4/files?cr_identifier=%s'
             % CatalogRecord.objects.get(pk=1).identifier)
         self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
 
+        # ... and should contain files ALL BUT THIS CR
+        response = self.client.get('/rest/directories/4/files?not_cr_identifier=%s'
+            % CatalogRecord.objects.get(pk=1).identifier)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.data['directories']), 1)
+        self.assertEqual(len(response.data['files']), 5)
+
     def test_read_directory_for_catalog_record_recursively(self):
         """
-        Test query parameter 'cr_identifier' with 'recursive'.
+        Test query parameters 'cr_identifier' with 'recursive'.
         """
         response = self.client.get('/rest/directories/1/files?recursive&cr_identifier=%s&depth=*'
             % CatalogRecord.objects.get(pk=1).identifier)
         self.assertEqual(response.status_code, status.HTTP_200_OK)
         self.assertEqual(len(response.data), 2)
+        for f in response.data:
+            self.assertTrue(f['parent_directory']['id'], CatalogRecord.objects.get(pk=1).identifier)
 
         # not found cr_identifier should raise 400 instead of 404, which is raised when the
         # directory itself is not found. the error contains details about the 400
         response = self.client.get('/rest/directories/1/files?recursive&cr_identifier=notexisting')
         self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
 
+    def test_read_directory_for_not_catalog_record_recursively(self):
+        """
+        Test query parameters 'not_cr_identifier' with 'recursive'.
+        """
+        response = self.client.get('/rest/directories/1/files?recursive&not_cr_identifier=%s&depth=*'
+            % CatalogRecord.objects.get(pk=1).identifier)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.data), 18)
+        for f in response.data:
+            self.assertNotEqual(f['parent_directory']['id'], CatalogRecord.objects.get(pk=1).id)
```

Comment:
This assert makes no sense? "assert file parent_directory id does not equal catalogrecord id"?

Reply:
mixed up CatalogRecord and Directory. New code is more "accurate".